lib/compress/zstd_ldm.h

```diff
@@ -37,21 +37,58 @@ extern "C" {
  *  parameters), which stores the matched sequences. The "long distance"
  *  match is then stored with the remaining literals from the
  *  ZSTD_blockCompressor. */
-size_t ZSTD_compressBlock_ldm(
-        ldmState_t* ldms, ZSTD_matchState_t* ms, seqStore_t* seqStore, U32 rep[ZSTD_REP_NUM],
-        ZSTD_CCtx_params const* params, void const* src, size_t srcSize);
 
-size_t ZSTD_compressBlock_ldm_extDict(
-        ldmState_t* ldms, ZSTD_matchState_t* ms, seqStore_t* seqStore, U32 rep[ZSTD_REP_NUM],
-        ZSTD_CCtx_params const* params, void const* src, size_t srcSize);
+/**
+ * ZSTD_ldm_generateSequences():
+ *
+ * Generates the sequences using the long distance match finder.
+ * The sequences completely parse a prefix of the source, but leave off the last
+ * literals. Returns the number of sequences generated into `sequences`.
+ *
+ * NOTE: The source may be any size, assuming it doesn't overflow the hash table
+ * indices, and the output sequences table is large enough..
+ */
+size_t ZSTD_ldm_generateSequences(
+        ldmState_t* ldms, rawSeq* sequences, ZSTD_matchState_t const* ms,
+        ldmParams_t const* params, void const* src, size_t srcSize,
+        int const extDict);
+
+/**
+ * ZSTD_ldm_blockCompress():
+ *
+ * Compresses a block using the predefined sequences, along with a secondary
+ * block compressor. The literals section of every sequence is passed to the
+ * secondary block compressor, and those sequences are interspersed with the
+ * predefined sequences. Returns the length of the last literals.
+ * `nbSeq` is the number of sequences available in `sequences`. After the
+ * function call it is set to the number of sequences consumed. `sequences` may
+ * be modified to handle boundary conditions.
+ *
+ * NOTE: The source must be at most the maximum block size, but the predefined
+ * sequences can be any size, and may be longer than the block. In the case that
+ * they are longer than the block, the last sequences may need to be split into
+ * two. We handle that case correctly, and update `sequences` and `nbSeq`
+ * appropriately.
+ */
+size_t ZSTD_ldm_blockCompress(rawSeq* sequences, size_t* nbSeq,
+    ZSTD_matchState_t* ms, seqStore_t* seqStore, U32 rep[ZSTD_REP_NUM],
+    ZSTD_compressionParameters const* cParams, void const* src, size_t srcSize,
+    int const extDict);
+
 
 /** ZSTD_ldm_initializeParameters() :
  *  Initialize the long distance matching parameters to their default values. */
 size_t ZSTD_ldm_initializeParameters(ldmParams_t* params, U32 enableLdm);
 
 /** ZSTD_ldm_getTableSize() :
```

Comment:
Could you document that `ZSTD_ldm_getTableSize()` is able to detect if Long Range mode is triggered or not, and appropriately reports a size of `0` when it is not ? 
It seems to be a change of scope.